### ISSUE TITLE
[Enhancement] DateTime converter

### ISSUE THREAD

**Issue by Eastrall** [xamarin/XamarinCommunityToolkit#1444]
Hello Xamarin Community Toolkit team and community!
I love the project so far and we are already using it in our Xamarin.Forms application!

## Summary

I would like to contribute to this project and suggest a new converter for the `DateTime` structure.
The idea would be to use it to format `DateTime`s directly through the XAML, instead of having a `string` property in view models and make the transformation there.

## API Changes

This would be a new converter named `DateTimeConverter` that could take a `string` as parameter to describe the format of the date.

```xaml
<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:xct="http://xamarin.com/schemas/2020/toolkit">
    <ContentPage.Resources>
        <xct:DateTimeConverter x:Key="DateTimeConverter" /> <!-- New API! -->
    </ContentPage.Resources>

    <Grid>
        <!-- Date property is a DateTime from ViewModel -->
        <Label Text="{Binding Date, Converter={x:StaticResource DateTimeConverter}, ConverterParameter='dd/MM'}" />
    </Grid>
</ContentPage>
```

## Intended Use Case

As said before, this convert offers a way to format a `DateTime` property into the given format instead of declaring a `string` property in the view model and then use the `DateTime.ToString("format")` method. This would be handled directly inside the XAML.

It could be used to display any kind of dates in specific formats like, `dd/MM`, `yyyy/MM/dd`, etc...

## Who Will Do The Work?

- [x] I am willing to take this on myself
- [ ] Just putting this out there for someone to pick up

--

Let me know if you are interested by this feature. Looking forward to discuss this topic with you. 😄

**Comment by GUOLDEV** [comment]
Hi @Eastrall, is there a reason you can not use `StringFormat` as described [here](https://docs.microsoft.com/en-us/xamarin/xamarin-forms/app-fundamentals/data-binding/string-formatting)?

**Comment by Eastrall** [comment]
Indeed, the use of `StringFormat` would have the same result! Didn't thought about it.

I don't know if it can be useful in addition to the initial feature request, but maybe the converter could have a predefined set of date formats, such as Angular's `DatePipe` : https://angular.io/api/common/DatePipe#pre-defined-format-options

Which allows to write something like:
```xml
<!--  Displays 'M/d/yy, h:mm a' format -->
<Label Text="{Binding Date, Converter={x:StaticResource DateTimeConverter}, ConverterParameter='ShortDate'}" />
```
or with the associated `enum DateTimeFormat` :

```xml
<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:xct="http://xamarin.com/schemas/2020/toolkit">
    <ContentPage.Resources>
        <xct:DateTimeConverter x:Key="DateTimeConverter" />
    </ContentPage.Resources>

    <Grid>
        <!-- Date property is a DateTime from ViewModel -->
        <Label Text="{Binding Date, Converter={x:StaticResource DateTimeConverter}, ConverterParameter={xct:DateTimeFormat.ShortDate}}" />
    </Grid>
</ContentPage>
```

**Comment by maxkoshevoi** [comment]
There're predefined format values for that too: https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwAICYCMBYAUHgEQENgBTAFQEsBbUgAigHsB3OgXjuLKtoDoA5FgG48qTAE4AFACICIOtLqIGLXuUYBlYGEpQA5jILSAlMZG4xU6QgVK6K5ms3bdB6ybOiJMgGLzFykyO6lo6+r4e5pYyAGbyAN4ADCAxAL7SMA6eFt7SAOIJyXnpmUHZ0dJ6hSB6JVlRuQCy1Y11ZQ1WAPLVnW0s5bmM1Yx9zANWAErVE6PjMgDO1fOzHTLk1eQrXlagdEkgwFs5VgCq1SdHFRDVEJe5AJrV9ytAA=

**Comment by ahoke-cr** [comment]
@maxkoshevoi @Eastrall  there's an even more complete list [in the Microsoft Docs](https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-date-and-time-format-strings (including format strings for other .NET types). I agree the StringFormat Binding property is a much simpler way to accomplish the intended purpose of this DateTimeConverter, though using a ValueConverter does confer the benefit of being able to debug the conversion process.

**Comment by Eastrall** [comment]
Hi, sorry for the late response. Definetly agree with both of you. I believe the issue doesn't make sense anymore, since we can achieve the same result using the `StringFormat` property in the XAML.

Thank you for all your feedbacks! I have learned the existance of the `StringFormat` property on the XAML bindings. Thus, we removed all or `DateTimeConverter` from our code and replaced them with the correct `StringFormat` value.

Closing this issue. 😄
